[Bugfix] Eror messages from Zou would not be returned properly

stevenpince opened the pull request:
**Problem**
Zou returns json objects with the 'error' key instead of 'message'. 
As an example; See https://github.com/cgwire/zou/blob/7f217f6be8cbc6eb1b10d80ca465c823b0aede40/zou/app/blueprints/files/resources.py#L705

**Solution**
I've added a utility function that checks a given request for the presence of both 'message' and 'error' keys, so that the error message of Zou can be properly passed on.

I also updated the UnitTests to test the following scenario's: verifying that the error message raised matches the one given by Zou and properly handling the error message when attempting to create a duplicate working file.

frankrousseau commented:
Thank you for this addition. 

@EvanBldy we have to make Zou error message more consistent.

stevenpince commented:
I saw the check for Python 2.7 fail. This is most likely caused by the type hinting that I added. Given that 2.7 reached EOL in 2020, and that current versions of Maya also switched to Python3, it seem acceptable to no longer support 2.7.

If you want, I can remove the type hinting and see if that enable 2.7 support.
